frontend/packages/data-portal/app/components/Run/ViewerPage.tsx

```diff
@@ -164,11 +164,83 @@ const isTopBarVisible = () => {
   return viewer?.uiConfiguration?.showLayerPanel.value ?? false
 }
 
-function ViewerPage({ run }: { run: any }) {
+const buildDepositsConfig = (annotations: any, tomograms: any[]) => {
+  const config: any = {}
+  const layers = currentNeuroglancerState().layers || []
+  for (const tomogram of tomograms) {
+    const depositionId = tomogram.deposition.id
+    if (
+      !(depositionId in config) &&
+      layers.some(
+        (l) => l.type === 'image' && l.source.endsWith(tomogram.s3OmezarrDir),
+      )
+    ) {
+      config[depositionId] = [tomogram]
+    }
+  }
+  for (const annotation of annotations.edges.map((e: any) => e.node)) {
+    const depositionId = annotation.depositionId
+    const httpsPath = annotation.httpsMetadataPath
+      .replace('.json', '')
+      .split('/')
+      .slice(-2)
+      .join('-')
+    const layer = layers.find(
+      (l) =>
+        l.source.includes?.(httpsPath) || l.source.url?.includes(httpsPath),
+    )
+    if (!(depositionId in config)) {
+      config[depositionId] = [{ name: layer?.name, annotation }]
+    } else {
+      config[depositionId].push({ name: layer?.name, annotation })
+    }
+  }
+  return config
+}
+
+const isDepositionActivated = (depositionEntries: string[]) => {
+  const layers = currentNeuroglancerState().layers || []
+  return layers
+    .filter((l) => l.name && depositionEntries.includes(l.name))
+    .some((l) => !boolValue(l.archived, /* defaultValue =*/ false))
+}
+
+const toggleDepositions = (depositionEntries: string[]) => {
+  const isCurrentlyActive = isDepositionActivated(depositionEntries)
+  updateState((state) => {
+    const layers = state.neuroglancer?.layers || []
+    for (const layer of layers.filter(
+      (l) => l.name && depositionEntries.includes(l.name),
+    )) {
+      layer.archived = isCurrentlyActive
+      layer.visible = !isCurrentlyActive
+    }
+    return state
+  })
+}
+
+const toggleAllDepositions = () => {
+  updateState((state) => {
+    const layers = state.neuroglancer?.layers || []
+    const layersOfInterest = layers.filter((l) => l.type !== 'image')
+    let archived = layersOfInterest.some((l) =>
+      boolValue(l.archived, /* defaultValue =*/ false),
+    )
+    for (const layer of layersOfInterest) {
+      layer.archived = !archived
+      layer.visible = archived
+    }
+    return state
+  })
+}
+
+function ViewerPage({ run, tomograms }: { run: any; tomograms: any }) {
```

Comment:
I don't see a change in this PR to use the tomograms, should there be a modification to `view.runs.$id.tsx` ?